File: LondonFhirService.Core/Models/Orchestrations/Comparisons/ComparisonResult.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System.Collections.Generic;
using System.Text.Json.Serialization;
using LondonFhirService.Core.Models.Processings.ListEntryComparisons;

namespace LondonFhirService.Core.Models.Orchestrations.Comparisons
{
    public class ComparisonResult
    {
        [JsonPropertyName("correlationId")]
        public string CorrelationId { get; set; } = string.Empty;

        [JsonPropertyName("diffCount")]
        public int DiffCount { get; set; }

        [JsonPropertyName("diffs")]
        public List<DiffItem> Diffs { get; set; } = new();
    }
}

File: LondonFhirService.Manage.Client/src/models/foundations/comparisons/ComparisonResult.ts
import type { DiffItem } from "./DiffItem";

// The shape stored in FhirRecordDifference.DiffJson. Mirrors
// LondonFhirService.Core.Models.Orchestrations.Comparisons.ComparisonResult.
export type ComparisonResult = {
    correlationId: string;
    diffCount: number;
    diffs: DiffItem[];
};
